### PR TITLE
Add rmkeyboards/rk13fa_tkl

### PR DESCRIPTION
**Pull request by ryanstevensmith**
Adding the initial upload of the rk13fa_tkl, an F13 TKL keyboard to fit RMKeyboards custom cases

## Description
Added all the files required to build a default and via keyboard firmware file for the rk13fa_tkl

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ x] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rmkeyboards/rk13fa_tkl/config.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x1345
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/config.h:22]: This vendor ID is in use, and needs to be changed:
https://www.the-sz.com/products/usbid/index.php?v=0x1345&p=&n=

Anything about 0x4000 is ideal

File: keyboards/rmkeyboards/rk13fa_tkl/config.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+/* number of backlight levels */
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/config.h:70]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/rmkeyboards/rk13fa_tkl/keymaps/wkl/keymap.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/keymaps/wkl/keymap.c:49]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by ryanstevensmith — COMMENTED** [review]
Changes made

File: keyboards/rmkeyboards/rk13fa_tkl/config.h (hunk 1 above)
Comment (ryanstevensmith) [reply keyboards/rmkeyboards/rk13fa_tkl/config.h]: changed to 0x4040

File: keyboards/rmkeyboards/rk13fa_tkl/rk13fa_tkl.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+void eeconfig_init_kb(void) {  // EEPROM is getting reset!
+  rgblight_enable(); // Enable RGB by default
+  rgblight_sethsv(0, 255, 128);  // Set default HSV - red hue, full saturation, medium brightness
+  rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2); // set to RGB_RAINBOW_SWIRL by default
+
+  eeconfig_update_kb(0);
+  eeconfig_init_user();
+}
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/rk13fa_tkl.c:41]:
```suggestion
```

None of this is necessary, TBH. 

The LEDs can be handled by defines:
https://docs.qmk.fm/#/feature_led_indicators?id=configuration-options

As can the default rgb settings:
https://docs.qmk.fm/#/feature_rgblight?id=configuration

File: keyboards/rmkeyboards/rk13fa_tkl/readme.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+RK13FA_TKL
+===
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/readme.md:2]:
```suggestion
# RK13FA_TKL
```

File: keyboards/rmkeyboards/rk13fa_tkl/readme.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+Keyboard Maintainer: RM Keyboards, [ryanstevensmith](https://github.com/ryanstevensmith)  
+Hardware Supported: RK13FA_TKL ver 1.1  
+Hardware Availability: contact RM Keyboards
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/readme.md:6]:
```suggestion
* Keyboard Maintainer: RM Keyboards, [ryanstevensmith](https://github.com/ryanstevensmith)  
* Hardware Supported: RK13FA_TKL ver 1.1  
* Hardware Availability: contact RM Keyboards
```

File: keyboards/rmkeyboards/rk13fa_tkl/readme.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+     
+    /* Row pin configuration
+    * row: 0   1   2   3   4   5   6   7   8   9   10  11
+    * pin: B0  B1  B2  B3  D0  D1  F0  F1  D7  B4  D6  D4  (Rev 1)
+    */
+    
+The RK13FA_TKL PCB utilizes duplex matrix organization.
+
+    RGB Data Pin: D3
+
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/readme.md:31]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rmkeyboards/rk13fa_tkl/readme.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# RK13FA_TKL
+
+* Keyboard Maintainer: RM Keyboards, [ryanstevensmith](https://github.com/ryanstevensmith)  
+* Hardware Supported: RK13FA_TKL ver 1.1  
+* Hardware Availability: contact RM Keyboards
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/readme.md:7]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/rmkeyboards/rk13fa_tkl/readme.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+     
+     make rmkeyboards/rk13fa_tkl:default
+   
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## RK13FA_TKL hardware information
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/readme.md:13]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rmkeyboards/rk13fa_tkl/config.h [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4040
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    RMKeyboards
+#define PRODUCT         RK13FA_TKL
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/config.h:26]:
```suggestion
```

File: keyboards/rmkeyboards/rk13fa_tkl/info.json [hunk 11]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "RK13FA_TKL",
+    "url": "",
+    "maintainer": "RM Keyboards",
```
Comment (drashna) [inline keyboards/rmkeyboards/rk13fa_tkl/info.json:4]:
```suggestion
    "maintainer": "RM Keyboards",
    "manufacturer": "RM Keyboards",
    "usb": {
        "vid": "0x4040",
        "pid": "0x0001",
        "device_version": "0.0.1"
    },
```

**Review by drashna — CHANGES_REQUESTED** [review]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Pull request closed without merging**